### PR TITLE
 Grep -C command v1

### PR DESCRIPTION
**Pull request by 31673**
Write a program to implement grep -C command. The program should accept a string and it should print the number of occurrences of the pattern in the file.

I used command line argument in this program for take pattern and file name .After main function it will check if condition whether the argument is equal to 3 or not .If it's not equal to 3 it will print enter valid argument.In else part open the file in read mode and read character by character by fgetc() and store in buffer. when the fgetc() read space or new line it will goes to compare with the read string which store in buffer and pattern given by user using strcmp .if same string count will increase or while will continue until end of file.
  for(iterator = 0; iterator < 20; iterator++  ) {
     buffer[iterator] = '\0';
} index = 0; 
This for loop used for free up the buffer to store next work.

After while loop break ,it will print number of occurrences of the pattern in the file.

Final output  ( No.of time Tata in test.txt = 4)

### CODE REVIEW

**Comment by P-Prajwal** [comment]
Can you paste the terminal output as well?

File: grep.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    char buffer[20];
+    char ch;
+    int count = 0;
+    int index = 0;
+    int iterator;
+    
+    if (argc != 3) {
+        printf("Invalid argument\n");
+	printf("Enter in following format\n");
+	printf("./a.out pattern filename\n");
```
Comment (preeja398) [inline grep.c:16]: Where this pattern used?

File: grep.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		    }
+		    else {
+		        buffer[index] = ch;
+			index++;
+		    }
+		}
+		printf("No.of time %s in %s = %d\n", argv[1],argv[2], count);
+	  }
+    }
+    fclose(fp);
```
Comment (P-Prajwal) [inline grep.c:46]: Same, again, close file if it was opened.

Comment (preeja398) [inline grep.c:46]: If fp is null, do u need to close file?

File: grep.c (hunk 1 above)
Comment (31673) [reply grep.c:16]: pattern used in else if statement to compare with readed word in like file.txt

File: grep.c (hunk 2 above)
Comment (31673) [reply grep.c:46]: No need to close.

**Pull request closed without merging**